homeassistant/components/emoncms/__init__.py

```diff
@@ -21,6 +24,39 @@ async def async_setup_entry(hass: HomeAssistant, entry: EmonCMSConfigEntry) -> b
         entry.data[CONF_API_KEY],
         session=async_get_clientsession(hass),
     )
+    emoncms_unique_id = await emoncms_client.async_get_uuid()
+    message = f"uuid for entry {entry.entry_id} is {entry.unique_id}"
+    LOGGER.debug(message)
+    ent_reg = er.async_get(hass)
+    entry_entities = ent_reg.entities.get_entries_for_config_entry_id(entry.entry_id)
+    if emoncms_unique_id:
+        if entry.unique_id != emoncms_unique_id:
+            for entity in entry_entities:
+                if entity.unique_id.split("-")[0] == entry.entry_id:
+                    feed_id = entity.unique_id.split("-")[-1]
+                    LOGGER.debug(f"moving feed {feed_id} to hardware uuid")
+                    ent_reg.async_update_entity(
+                        entity.entity_id, new_unique_id=f"{emoncms_unique_id}-{feed_id}"
+                    )
+            hass.config_entries.async_update_entry(
+                entry,
+                data=entry.data,
+                unique_id=emoncms_unique_id,
+            )
+    else:
+        async_create_issue(
```

Comment:
Is the issue removed automatically after a database upgrade?

Reply:
Once the user updates its emoncms database, the issue is removed after a home-assistant restart